language/Readme.md

```diff
@@ -33,4 +33,13 @@ already compiled in the `target/` folder. A simple `cargo clean` followed by
 ## Testing
 
 The compiler can be tested by launching `cargo test`. It typechecks and compile to F* all 
-the specs in [examples/](../examples/).
\ No newline at end of file
+the specs in [examples/](../examples/).
+
+## Updating exported files
+To be able to update exported files you should use the `--init` flag when you first 
+build you files, this will create a `<output>_template` file with the historical view. 
```

Comment:
What does "historical view" mean ?

Reply:
How the file looked before proofs or other additions were made, so the clean/new build file is a "historical view" compared to the file you work on.

Reply:
Ok there's something I don't understand here. The git merge workflow I used for F* was the following. Let's say you have `A.rs` in Rust and translate it with hacspec in `A.fst`. Then you modify `A.fst` with annotations. But then you tweak `A.rs` so you have to re-generate `A.fst`.

In this case I generate `A.fst.tmp` which corresponds to the new version of `A.rs`, and I git-merge it with `A.fst` corresponding to the old version of `A.rs` but with my annotations.

If I understand correctly, here you want to always keep *two* `A.fst` file, one "bare" straight from the compiler ("historical view") and the other where you put your corrections. But what is the purpose of the "historial view", since it will be overwritten by the hacspec compiler each time you modify `A.rs` ? 

I'm curious because your logic in `main.rs` also creates a temporary file and copies it over.

Reply:
Yes, I keep two versions one bare `A_template.v` and one work file `A.v`. So when doing `--init` I get two identical files, then I would work in one `A.v`, and when I make any changes to `A.rs` I will make a new bare file `A_temp.v` which I use `git merge-file` to compare to the historical file `A_template.v` and the work file `A.v`. After the merge I will update the template file `A_template.v` by  replacing it with the new bare file `A_temp.v`. The reason for the extra file is to be able to make any changes not just incremental once in the work file, and then get merge conflicts if there are any issues. I hope this will be better then just appending new changes.

Reply:
Ah okay I see keeping the `A_template.v` file lets you do a three-way merge instead of a two-way merge, which helps git's merge algorithm to be more spot on with the changes. This is clever!